src/main/java/org/embulk/input/gcs/GcsFileInput.java

```diff
@@ -80,17 +80,25 @@ static FileList listFiles(PluginTask task)
         return builder.build();
     }
 
-    // String nextToken = base64Encode(0x0a + 0x01~0x27 + filePath);
+    // String nextToken = base64Encode(0x0a + ASCII character according to utf8EncodeLength position+ filePath);
     @VisibleForTesting
     static String base64Encode(String path)
     {
         byte[] encoding;
         byte[] utf8 = path.getBytes(Charsets.UTF_8);
         LOG.debug("path string: {} ,path length:{} \" + ", path, utf8.length);
 
+        int utf8EncodeLength = utf8.length;
+        if (utf8EncodeLength >= 128) {
+            throw new ConfigException("Can't support last path encode length longer than 127. Please try to reduce the length");
```

Comment:
none blocking but can you explain more which length users should reduce?

Reply:
Updated

Reply:
@hieudion , in my opinion this last_path value get from file (bucket+ path_prefix + filename) from gcs and this config is invisible to user so they can miss understand what is last_path. How do you think if we show a message like "Some file name from your gcs have name more than .... char. Please reduce file name character." From our code we can check some fix value for ex: bucket lenght + patt_prefix... So we can know exactly how many characters is ok for file name.